server/server.go

```diff
@@ -31,8 +31,9 @@ type Server struct {
 
 // Start initializes a server to respond to HTTPS/TLS network requests.
 func (s *Server) Start(ctx context.Context) error {
-	// Uncomment the following line for local run
-	// s.Insecure = true
+	// The default run mode is insecure, as most clients will run the delegate and
+	// the docker runner on a same host.
+	s.Insecure = true
```

Comment:
You're hardcoding insecure=true which means if someone needs to run with TLS, you'll still run it in insecure mode. My suggestion would be to change the default value in the config to be true. You can make this change here: https://github.com/harness/harness-docker-runner/blob/57cbb7fcc5f512b01cb290bfd5488a2b0b024605/config/config.go#L23

cc @shubham149

Reply:
What you pointed out is valid, though the default value should be true.